services/api/src/api/authentication.py

```diff
@@ -35,10 +36,25 @@ def __call__(self, r: PreparedRequest) -> PreparedRequest:
         return r
 
 
+def is_jwt_valid(request: Optional[Request], public_key: Optional[str]) -> bool:
+    if not request or not public_key:
+        return False
+    encoded = request.headers.get("x-api-key")
+    if not encoded:
+        return False
+    try:
+        jwt.decode(jwt=encoded, key=public_key, algorithms=["RS256"])
+        # ^ we're not interested in the contents of the JWT token, we just want to check that it's valid
```

Comment:
I strongly suggest you also check the JWT content, otherwise any JWT generated by moon-landing will be invalid

Reply:
eg, the JWT passed to Spaces' iframes

Reply:
OK, you mean that somebody could reuse a JWT they got in their space?

Reply:
Done